staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/types.go

```diff
@@ -134,3 +137,54 @@ func (q *queue) dump(includeDetails bool) debug.QueueDump {
 		SeatsInUse:        q.seatsInUse,
 	}
 }
+
+// SeatSeconds is a measure of work, in units of seat-seconds, using a fixed-point representation.
+// `SeatSeconds(n)` represents `n/ssScale` seat-seconds.
+// The constants `ssScale` and `ssScaleDigits` are private to the implementation here,
+// no other code should use them.
+type SeatSeconds uint64
+
+// MaxSeatsSeconds is the maximum representable value of SeatSeconds
+const MaxSeatSeconds = SeatSeconds(math.MaxUint64)
+
+// SeatsTimesSeconds produces the SeatSeconds value for the given factors.
+// This is intended only to produce small values, increments in work
+// rather than amount of work done since process start.
+func SeatsTimesSeconds(seats, seconds float64) SeatSeconds {
+	return SeatSeconds(math.Round(seats * seconds * ssScale))
+}
+
+// SeatsTimeDuration produces the SeatSeconds value for the given factors.
+// This is intended only to produce small values, increments in work
+// rather than amount of work done since process start.
+func SeatsTimesDuration(seats float64, duration time.Duration) SeatSeconds {
+	return SeatSeconds(math.Round(seats * float64(duration/time.Nanosecond) / (1e9 / ssScale)))
+}
+
+// ToFloat converts to a floating-point representation.
+// This conversion may lose precision.
+func (ss SeatSeconds) ToFloat() float64 {
```

Comment:
Same here - can we get rid of this function?

Reply:
No, because we need floating-point values to set Prometheus gauges.